### PR TITLE
Init spring maven plugin

### PR DESCRIPTION
**Pull request by Flanker32**
Init spring maven plugin and deploy mojo.

### CODE REVIEW

File: azure-spring-maven-plugin/pom.xml [hunk 1]
```diff
@@ -0,0 +1,228 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <groupId>com.ms.azure</groupId>
```
Comment (andxu) [inline azure-spring-maven-plugin/pom.xml]: com.microsoft.azure

File: azure-spring-maven-plugin/pom.xml [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+    <parent>
+        <groupId>com.microsoft.azure</groupId>
+        <artifactId>azure-maven-plugins-pom</artifactId>
+        <version>1.0.2-SNAPSHOT</version>
+        <relativePath>../azure-maven-plugins-pom</relativePath>
+    </parent>
+    <name>Azure Spring Cloud Maven Plugin</name>
+
+    <!-- FIXME change it to the project's website -->
+    <url>http://www.example.com</url>
```
Comment (andxu) [inline azure-spring-maven-plugin/pom.xml]: fix it

File: azure-spring-maven-plugin/pom.xml [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>com.microsoft.azure</groupId>
+            <artifactId>azure-storage-blob</artifactId>
+            <version>11.0.1</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.taobao.text</groupId>
```
Comment (andxu) [inline azure-spring-maven-plugin/pom.xml]: remove it

File: azure-spring-maven-plugin/pom.xml [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+            <artifactId>text-ui</artifactId>
+            <version>0.0.2</version>
+        </dependency>
+
+        <!-- https://mvnrepository.com/artifact/org.apache.maven.shared/maven-shared-utils -->
```
Comment (andxu) [inline azure-spring-maven-plugin/pom.xml]: remove this line

File: azure-spring-maven-plugin/pom.xml [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+        </dependency>
+
+
+        <dependency>
+            <groupId>com.microsoft.azure.microservices4spring.v2019_05_01_preview</groupId>
+            <artifactId>azure-mgmt-microservices4spring</artifactId>
+            <version>1.0.0-beta</version>
+        </dependency>
+
+        <dependency>
```
Comment (andxu) [inline azure-spring-maven-plugin/pom.xml]: remove this lib

File: azure-spring-maven-plugin/pom.xml [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-plugin-api</artifactId>
+            <version>${maven.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>com.microsoft.azure</groupId>
+            <artifactId>azure</artifactId>
+            <version>1.21.0</version>
```
Comment (andxu) [inline azure-spring-maven-plugin/pom.xml]: use version in properties.

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java [hunk 7]
```diff
@@ -0,0 +1,111 @@
...
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public class Utils {
+
+    private static final String DATE_FORMAT = "yyyyMMddHHmmss";
+    private static final String MEMORY_REGEX = "(\\d+(\\.\\d+)?)([a-zA-Z]+)";
+    private static final Pattern MEMORY_PATTERN = Pattern.compile(MEMORY_REGEX);
+
+    private Utils() {
```
Comment (andxu) [inline azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: move to the last

File: azure-spring-maven-plugin/pom.xml (hunk 1 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/pom.xml]: Fixed

File: azure-spring-maven-plugin/pom.xml (hunk 3 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/pom.xml]: Done

File: azure-spring-maven-plugin/pom.xml (hunk 2 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/pom.xml]: Done

File: azure-spring-maven-plugin/pom.xml (hunk 4 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/pom.xml]: Done

File: azure-spring-maven-plugin/pom.xml (hunk 6 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/pom.xml]: Done

File: azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java (hunk 7 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/utils/Utils.java]: Done

File: azure-spring-maven-plugin/pom.xml (hunk 5 above)
Comment (Flanker32) [reply azure-spring-maven-plugin/pom.xml]: Done

**Pull request merged**